lib/improver/tests/utilities/test_cube_manipulation.py

```diff
@@ -1297,5 +1301,154 @@ def test_multiple_cell_methods_in_input_cube(self):
+                         result.coord_dims(coord_name))
+        self.assertArrayAlmostEqual(
+            self.descending_height_points, result.coord(coord_name).points)
+        self.assertDictEqual(
+            result.coord(coord_name).attributes, {"positive": "down"})
+        self.assertArrayAlmostEqual(result.data, expected_data)
+
+    def test_descending_then_ascending(self):
+        """Test that the sorting successfully sorts the cube based
+        on the points within the given coordinate. The points in the resulting
+        cube should now be in ascending order."""
+        expected_data = np.array(
+            [[[[3.00, 3.00, 3.00],
+               [3.00, 3.00, 3.00],
+               [3.00, 3.00, 3.00]]],
+             [[[2.00, 2.00, 2.00],
+               [2.00, 2.00, 2.00],
+               [2.00, 2.00, 2.00]]],
+             [[[1.00, 1.00, 1.00],
+               [1.00, 1.00, 1.00],
+               [1.00, 1.00, 1.00]]]])
+        coord_name = "height"
+        result = sort_coord_in_cube(self.descending_cube, coord_name)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(self.ascending_cube.coord_dims(coord_name),
+                         result.coord_dims(coord_name))
+        self.assertArrayAlmostEqual(
+            self.ascending_height_points, result.coord(coord_name).points)
+        self.assertDictEqual(
+            result.coord(coord_name).attributes, {"positive": "up"})
+        self.assertArrayAlmostEqual(result.data, expected_data)
+
+    def test_descending_then_descending(self):
+        """Test that the sorting successfully sorts the cube based
+        on the points within the given coordinate. The points in the resulting
+        cube should now be in descending order."""
+        expected_data = np.array(
+            [[[[1.00, 1.00, 1.00],
+               [1.00, 1.00, 1.00],
+               [1.00, 1.00, 1.00]]],
+             [[[2.00, 2.00, 2.00],
+               [2.00, 2.00, 2.00],
+               [2.00, 2.00, 2.00]]],
+             [[[3.00, 3.00, 3.00],
+               [3.00, 3.00, 3.00],
+               [3.00, 3.00, 3.00]]]])
+        coord_name = "height"
+        result = sort_coord_in_cube(
+            self.descending_cube, coord_name, order="descending")
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(self.descending_cube.coord_dims(coord_name),
+                         result.coord_dims(coord_name))
+        self.assertArrayAlmostEqual(
+            self.descending_height_points, result.coord(coord_name).points)
+        self.assertDictEqual(
+            result.coord(coord_name).attributes, {"positive": "down"})
+        self.assertArrayAlmostEqual(result.data, expected_data)
+
+    def test_latitude(self):
+        """Test that the sorting successfully sorts the cube based
+        on the points within the given coordinate (latitude).
+        The points in the resulting cube should now be in descending order."""
```

Comment:
What will happen if a longitude coordinate is given that spans the meridian? Should we have a check in the code to make sure there aren't any discontinutities like this? It might well be out-of-scope for IMPROVER.

Reply:
An interesting thought. At the moment, I'm decided to capture this by raising a warning if the coordinate is `circular`. However, ideally this is something that we could handle, although it might be sufficiently complicated that it could be handled in a separate PR when the functionality is necessary.